### PR TITLE
Switch to the last Watermill version

### LINKED ISSUE #61: Switch to the last Watermill version
As we applied a temporary workaround and reported a new issue to Watermill project. It's a reminder, to monitor for the fix and to switch to the last version when it's ready. Details are in:
- #58.

### PR DESCRIPTION
**Pull request by daniel-milchev**
[#61] Switched to the latest version of watermill.
Signed-off-by: Daniel Milchev fixed-term.daniel.milchev@bosch.io

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -4,18 +4,18 @@ go 1.17
 
 require (
 	github.com/Jeffail/gabs/v2 v2.6.0
-	github.com/ThreeDotsLabs/watermill v1.1.1
+	github.com/ThreeDotsLabs/watermill v1.3.2
 	github.com/caarlos0/env/v6 v6.10.1
-	github.com/cenkalti/backoff/v3 v3.0.0
+	github.com/cenkalti/backoff/v3 v3.2.2
```
Comment (dimitar-dimitrow) [inline go.mod:9]: The cenkalti/backoff/v3 version has increased also, but it is not updated in the NOTICE file. Use _go version -m path-to-suite-connector-binary_ to list the versions of the build dependencies that goes in the binary and update them in the NOTICE file.

**Review by dimitar-dimitrow — CHANGES_REQUESTED** [review]

**Comment by k-gostev** [comment]
Did anyone test if the build is working with go 1.17.0? Yocto recipes should be verified as well before merging.

**Comment by hristobojilov** [comment]
This build now requires go 1.20 but yocto should build successfully.

File: go.mod [hunk 2]
```diff
@@ -4,18 +4,18 @@ go 1.17
 
 require (
 	github.com/Jeffail/gabs/v2 v2.6.0
-	github.com/ThreeDotsLabs/watermill v1.1.1
```
Comment (hristobojilov) [inline go.mod:7]: Update go version used to 1.20

**Review by hristobojilov — CHANGES_REQUESTED** [review]

**Review by hristobojilov — APPROVED** [review]

**Review by dimitar-dimitrow — APPROVED** [review]

**Pull request merged**